libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1001,9 +1002,9 @@ void TypeChecker::endVisit(TryStatement const& _tryStatement)
 	TryCatchClause const* panicClause = nullptr;
 	TryCatchClause const* errorClause = nullptr;
 	TryCatchClause const* lowLevelClause = nullptr;
-	for (size_t i = 1; i < _tryStatement.clauses().size(); ++i)
+	for (auto const& clausePointer: _tryStatement.clauses() | ranges::views::drop_exactly(1))
 	{
-		TryCatchClause const& clause = *_tryStatement.clauses()[i];
+		TryCatchClause const& clause = *clausePointer;
```

Comment:
It's not like we really have to dereference ``clausePointer`` to ``clause``, that would make it shorter.

Or we could just define additional views in ``CommonData.h`` (or so), i.e.

```
namespace views
{
static constexpr auto dereference= ranges::views::transform([](auto*  x)  { return *x; });
}
```

with that this would become:
```

for (TryCatchClause const& clause: _tryStatement.clauses() | ranges::views::drop_exactly(1) | views::dereference)
{
```

Also what we could do is to try to use ``ranges::cpp20`` versions, where applicable, then this could become, here that'd be ``ranges::cpp20::views::drop(1)``. We could even consider
```
namespace views
{
  using namespace ranges::cpp20::views;
}
```
Then we could use ``views::drop(1)`` and once we can switch to C++20, we only have to change that to
```
namespace views
{
   using namespace std::ranges::views;
}
```
But not sure in which header to put that and how to make sure that ``ranges::cpp20::views`` is already defined in it (although we could just define it ourselves as empty).


All that being said, I'd say this is already nicer as before as it is, though :-).

Reply:
Maybe we could have a new header that just includes all the commonly used stuff from `ranges`? Currently we have to include headers like `range/v3/view/drop_exactly.hpp` everywhere so having ours with some extra definitions would not be worse that what we are doing currently.

And yeah, `ranges::cpp20::views::drop(1)` is ridiculous :) Having a custom namespace to shorten that would be a great usability improvement. I'd even go as far as
```c++
using ranges::cpp20::views::drop;
```
but only if that was in the same file where I use the function. Not necessarily a good idea if hidden in a helper header.

Reply:
I'd be careful with a common header pulling in a lot of commonly used ranges - it can rather quickly hurt compile-time significantly...
The namespace import in a ``views`` namespace of our own in some header would mainly be for making the replacement with the ``std`` versions more straightforward in the long run (not that we wouldn't still need to adjust the headers, but well) - you could still use ``using views::drop`` or even ``using views;`` in your implementation, if you want :-)...

Reply:
Ah, right. It's header-only so it all needs recompilation every time.